### PR TITLE
Fix a bug that training is stuck while detection model is trained on distrubited environment

### LINKED ISSUE #3635: Problems in multi-card distributed training
**The problem of distributed training blocking**

**Steps to Reproduce**

1、Minimum code block

from otx.engine import Engine

engine = Engine(model="yolox_s",  data_root="pwd")
engine.train(num_nodes=2) 

2.I tried other code to troubleshoot my environment.

import lightning as L
from lightning.pytorch.demos.boring_classes import BoringModel

ngpus = 2
model = BoringModel()
trainer = L.Trainer(max_epochs=10,
                    devices=ngpus)

trainer.fit(model)

log:
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/2
Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/2
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 2 processes

**Environment:**

- OS: 
- Framework version: 
- Python version: 3.10
- OpenVINO version:
- CUDA/cuDNN version: 12.2
- GPU model and memory: 24G(4090)*2

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
This PR fixes #3635

### How this change solves the bug
In distributed training, torchmetric package tries to sync values across processes.
If one process has a value in bbox, label or scores and other process not, then torchmetric makes empty torch Tensor.
But if dtype of tensor between processes is different, then process is stuck during sync.
To avoid it, this PR specifies dtype properly.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/data/dataset/detection.py [hunk 1]
```diff
@@ -49,8 +49,9 @@ def _get_item_impl(self, index: int) -> DetDataEntity | None:
                 bboxes,
                 format=tv_tensors.BoundingBoxFormat.XYXY,
                 canvas_size=img_shape,
+                dtype=torch.float32,
             ),
-            labels=torch.as_tensor([ann.label for ann in bbox_anns]),
+            labels=torch.as_tensor([ann.label for ann in bbox_anns], dtype=torch.int32),
```
Comment (sungchul1) [inline src/otx/core/data/dataset/detection.py]: Could you align it with iseg to int64 if it works?

Comment (eunwoosh) [reply src/otx/core/data/dataset/detection.py]: I think it should work but is it needed to set it to torch.int64? I think there is no case to use more labels than upper limit of int32.

Comment (eunwoosh) [reply src/otx/core/data/dataset/detection.py]: It seems that some code in detection head is using `torch.long` for label calculation and need to align it with that. maybe this is why iseg uses torch.int64. Anyway, I updated it to `torch.long`. Thanks.

Comment (sungchul1) [reply src/otx/core/data/dataset/detection.py]: I think so, otx seems to use double and just aligning it :)

**Review by sungchul1 — APPROVED** [review]

**Review by harimkang — APPROVED** [review]
